### ISSUE TITLE
Not working for Typescript Angular 7 project

### ISSUE THREAD

**Issue by zentoaku** [depcheck/depcheck#319]
Hello,

I'm trying to figure out how to use this tool with Angular 7 project which uses Typescript. I've seen that there should be a built-in support for it but so far the tool is reporting all of my dependencies from package.json as "unused". 
Tried:
> depcheck ./
> depcheck ./ --parsers="*.ts:ts"
> depcheck ./ --parsers="*.ts:typescript"

Could you give me some more clarity on how to use this tool from command line?

**Comment by rumpl** [comment]
Hi @zentoaku,

I just tried this:
* install ng-cli: `npm install -g @angular/cli`
* create a new app: `ng new my-app`
* run depcheck: `depcheck .`

And this is the output I get: 
```
Unused dependencies
* @angular/animations
* @angular/forms
* core-js
* tslib
Unused devDependencies
* @angular/cli
* @angular/compiler-cli
* @angular/language-service
* @types/node
* @types/jasmine
* @types/jasminewd2
* codelyzer
* tslint
* typescript
```

The unused dependencies are correct as for the devDependencies, there is a bunch of stuff going on:
* the `@types/*` are not yet supported by depcheck, should be there soon
* not sure about the cli or compiler-cli
* tslint, typescript and codelyzer seem like a bug

Could you perhaps share a repo against which you are running depcheck? Or at least give us the output of `depcheck . --json`.

Thanks

**Comment by zentoaku** [comment]
I can't share the repo but here is the command output:
```
{
  "dependencies": [
    "@angular/animations",
    "@angular/common",
    "@angular/compiler",
    "@angular/core",
    "@angular/forms",
    "@angular/http",
    "@angular/platform-browser",
    "@angular/platform-browser-dynamic",
    "@angular/router",
    "@xxx/breadcrumbs",
    "@xxx/cache",
    "@xxx/globals",
    "@xxx/notification",
    "@xxx/security",
    "@xxx/translation",
    "@xxx/web-storage",
    "@ng-bootstrap/ng-bootstrap",
    "@ngrx/effects",
    "@ngrx/router-store",
    "@ngrx/store",
    "@ngrx/store-devtools",
    "@ngx-translate/core",
    "@progress/kendo-angular-buttons",
    "@progress/kendo-angular-dateinputs",
    "@progress/kendo-angular-dropdowns",
    "@progress/kendo-angular-excel-export",
    "@progress/kendo-angular-grid",
    "@progress/kendo-angular-inputs",
    "@progress/kendo-angular-intl",
    "@progress/kendo-angular-l10n",
    "@progress/kendo-data-query",
    "@progress/kendo-drawing",
    "@progress/kendo-theme-bootstrap",
    "angular-oauth2-oidc",
    "bootstrap",
    "core-js",
    "font-awesome",
    "ngx-toastr",
    "rxjs",
    "rxjs-compat",
    "tslib",
    "zone.js"
  ],
  "devDependencies": [
    "@angular/compiler-cli",
    "@angular/language-service",
    "@types/jasmine",
    "@types/jasminewd2",
    "@types/lodash",
    "@types/node",
    "codelyzer",
    "enhanced-resolve",
    "jasmine-core",
    "karma",
    "karma-chrome-launcher",
    "karma-coverage-istanbul-reporter",
    "karma-jasmine",
    "karma-jasmine-html-reporter",
    "ngrx-store-freeze",
    "tslint",
    "tslint-sonarts",
    "typescript"
  ],
  "missing": {},
  "using": {
    "jasmine-spec-reporter": [
      "C:\\projects\\pos-angular\\e2e\\protractor.conf.js"
    ],
    "ts-node": [
      "C:\\projects\\pos-angular\\e2e\\protractor.conf.js"
    ],
    "colors": [
      "C:\\projects\\pos-angular\\generate-version.js"
    ],
    "@angular-devkit/build-angular": [
      "C:\\projects\\pos-angular\\karma.conf.js"
    ],
    "karma-sonarqube-reporter": [
      "C:\\projects\\pos-angular\\karma.conf.js"
    ],
    "@angular/cli": [
      "C:\\projects\\pos-angular\\package.json"
    ],
    "protractor": [
      "C:\\projects\\pos-angular\\package.json"
    ]
  },
  "invalidFiles": {
    "C:\\projects\\pos-angular\\karma.conf.js": "Error: Cannot find module 'karma-*' from 'C:\\projects\\pos-angular'\n    at Function.module.exports [as sync] (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\node_modules\\resolve\\lib\\sync.js:58:15)\n    at karmaPluginsInstalled.forEach.plugin (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\special\\karma.js:91:42)\n    at Array.forEach (<anonymous>)\n    at collectInstalledPluginInfo (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\special\\karma.js:90:25)\n    at collectUsages (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\special\\karma.js:167:22)\n    at parseKarma (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\special\\karma.js:183:10)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\app\\core\\components\\footer\\footer.component.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\app\\core\\components\\footer\\_variables.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\app\\core\\components\\navigation-bar\\navigation-bar-menu-item\\navigation-bar-menu-item.component.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\app\\core\\components\\navigation-bar\\navigation-bar-menu-item\\_variables.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\app\\core\\components\\navigation-bar\\navigation-bar.component.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\app\\core\\components\\navigation-bar\\_variables.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\app\\core\\services\\notification\\notification.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\app\\features\\parts\\containers\\parts-page\\parts-page.component.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\app\\features\\parts\\components\\parts-grid\\parts-grid.component.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\app\\features\\parts\\components\\parts-search\\parts-search.component.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\styles.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)",
    "C:\\projects\\pos-angular\\src\\_variables.scss": "TypeError: Cannot read property 'renderSync' of null\n    at parseSASS (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\parser\\sass.js:21:33)\n    at _fs.default.readFile (C:\\Users\\xxx\\AppData\\Roaming\\npm\\node_modules\\depcheck\\dist\\check.js:77:17)\n    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)"
  },
  "invalidDirs": {}
}
```

**Comment by k3yb0ardn1nja** [comment]
I am experiencing the same issue. I create a sample to exhibit the behavior. It doesn't seem to be a package-dependent issue, so I just picked on `rxjs`. The only change from the initial Angular 7 CLI `new` see was adding a simple use of `rxjs` in `src/app/app.component.ts`:
https://github.com/k3yb0ardn1nja/ng7-cli-depcheck-test

Any guidance is greatly appreciated!

**Comment by rumpl** [comment]
not stale

**Comment by Bilalm354** [comment]
not stalw

**Comment by tony** [comment]
Hey this isn't stale.  For those who have depcheck working, can you please paste the `depcheck` command and/or configuration file(s) please? thank you!

**Comment by rumpl** [comment]
Nothing is really ever stale and all the issues should stay open, the truth is I do not have enough time to fix all the issues, new frameworks and new ways of importing code comes out every day. I am open to contributions though so if you feel like this is something that should *really* be fixed/implemented then I'll be more than happy to help with reviews or hand-holding around the codebase.

**Comment by boeckMt** [comment]
> Hey this isn't stale. For those who have depcheck working, can you please paste the `depcheck` command and/or configuration file(s) please? thank you!

Hey I use it inside a script 


```ts
import * as depcheck from 'depcheck';
...




const results = await depcheck(path, options);

```

**Comment by tony** [comment]
@rumpl 

Hey, thank you for the project!  This has a lot of work put into it.

>  the truth is I do not have enough time to fix all the issues, new frameworks and new ways of importing code comes out every day.

Yes it seems hard to comprehend the challenge - The myriad of ways JS projects can be assembled, invoked, etc. is immense.

>  I am open to contributions though so if you feel like this is something that should really be fixed/implemented then I'll be more than happy to help with reviews or hand-holding around the codebase.

Yes you're also right here.

Hey I want to make a side note: It's easier said than done to make a PR, there's a learning curve to understand the innards of a project and contribute to it.  If they haven't used it on a simpler project, they may not know how effective depcheck would be for their case

@boeckMt  Thank you!

How is this invoked? Is it invoked as part of a normal node / `ts-node` script?

**Comment by rumpl** [comment]
@boeckMt what version of depcheck are you using? I think this line:
```
  parsers: { // the target parsers
    '*.ts': depcheck.parser.typescript
  },
```

Should be 

```
  parsers: { // the target parsers
    '**/*.ts': depcheck.parser.typescript
  },
```

**Comment by rumpl** [comment]
@tony Thank you for understanding :)

> Hey I want to make a side note: It's easier said than done to make a PR, there's a learning curve to understand the innards of a project and contribute to it. If they haven't used it on a simpler project, they may not know how effective depcheck would be for their case

I do realize this, my hope is that one day I'll have the time and courage to fix some of the more complicated things or that someone with a will does it for everyone :)

**Comment by boeckMt** [comment]
> @boeckMt what version of depcheck are you using? I think this line:
> 
> ```
>   parsers: { // the target parsers
>     '*.ts': depcheck.parser.typescript
>   },
> ```
> 
> Should be
> 
> ```
>   parsers: { // the target parsers
>     '**/*.ts': depcheck.parser.typescript
>   },
> ```

@rumpl I'm using version `^1.4.0`
All details are in the package https://github.com/dlr-eoc/ukis-frontend-libraries/blob/v7.3.1/scripts/library and it is used inside the file https://github.com/dlr-eoc/ukis-frontend-libraries/blob/v7.3.1/scripts/library/utils.ts

**Comment by rumpl** [comment]
Hm, I'm surprised it works then, we changed the way regexes work quite a while ago

**Comment by boeckMt** [comment]
> Hm, I'm surprised it works then, we changed the way regexes work quite a while ago

@rumpl you are right, I don't have recognised that it was not working as expected.

So it really should be

```ts
parsers: { // the target parsers
    '**/*.ts': depcheck.parser.typescript
  },
```
